### ISSUE TITLE
Support for Umbraco Heartcore

### ISSUE THREAD

**Issue by tednyberg** [weareyou/gatsby-source-umbraco#15]
Perhaps this is more a question than a suggestion, but is there (or will there be) support for Umbraco Heartcore?

In some ways I think it would be easier to support Heartcore than Headrest, given that the Heartcore API is forcibly more consistent/predictable than Headrest implementations?

**Comment by jessevdp** [comment]
Hi, great question(s)! 👌

A Gatsby source plugin is essentially an adapter. Its goal is to take data from a source, transform it where necessary, and hand it to Gatsby in the correct format. This source plugin was written to pull data from (regular old) Umbraco. But pulling the data is only half the problem, it also needs to be exposed via a (custom) API. Because regular Umbraco doesn’t have an API, it will have to be custom built. This plugin requires the API to [follow some conventions](https://github.com/weareyou/gatsby-source-umbraco#api-design). 

**Note that this plugin does *not* exclusively work with HeadRest**. It supports any API on top of Umbraco—as long as it can comply with the minimal conventions. It just so happens that implementing such an API with HeadRest is easy. It pairs well with this plugin.

Since Umbraco Hearthcore does not follow the same format, this plugin doesn't (and won't) support it. Changing the (rather minimal) conventions that this plugin requires to something more complicated (and bloated) as the HAL format that Heartcore's API uses would make this plugin much less useful. Especially because...

It is true that it would be fairly easy to create a similar Gatsby plugin to pull data from Umbraco Heartcore; the fact that it has an API already solves half of the problem. Their SDK should make it even easier. However, **it should not be necessary to create a custom plugin**. Heartcore will get support for GraphQL "soon" (see the [pricing page](https://umbraco.com/umbraco-heartcore-pricing/)), and there is already a Gatsby plugin to pull content from any GraphQL API (see [gatsby-source-graphql](https://www.gatsbyjs.org/packages/gatsby-source-graphql/)).

**In short**—No this plugin doesn't and won't support Umbraco Heartcore. It has a different purpose. But, you should be able to connect Umbraco HeartCore to Gatsby by using [gatsby-source-graphql](https://www.gatsbyjs.org/packages/gatsby-source-graphql/).

Hope this helps! 🙌

**Comment by tednyberg** [comment]
Makes perfect sense, thanks!
